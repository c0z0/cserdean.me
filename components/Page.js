import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import Head from 'next/head';

import Menu from './Menu';
import * as breakPoints from '../utils/breakPoints';

const Container = styled.div`
  font-size: 12px;
  color: ${({ textColor }) => textColor};
  font-family: 'SF Pro Text', -apple-system, BlinkMacSystemFont, 'Segoe UI',
    Roboto, Helvetica, Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji',
    'Segoe UI Symbol';
  max-width: 900px;
  margin: 0 auto;
`;

const Content = styled.div`
  margin: 80px 0;

  @media (${breakPoints.phoneOnly}) {
    margin: 40px 20px;
  }
`;

export default function Page({ children, active }) {
  return (
    <Container>
      <Head>
        <title>Cosmin Serdean</title>
      </Head>
      <Menu active={active} />
      <Content>{children}</Content>
    </Container>
  );
}

Page.propTypes = {
  children: PropTypes.node.isRequired,
  active: PropTypes.string.isRequired,
};
