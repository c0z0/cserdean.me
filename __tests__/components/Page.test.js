import React from "react";
import { render, cleanup } from "react-testing-library";
import Router from "next/router";

import Page from "../../components/Page";

const mockedRouter = {
  push: jest.fn(async () => {}),
  prefetch: () => {}
};
Router.router = mockedRouter;

afterEach(cleanup);

test("Matches snapshot", () => {
  const { container, rerender } = render(<Page active="/">test content</Page>);

  expect(container.firstChild).toMatchSnapshot();

  rerender(
    <Page active="/" dark>
      test content
    </Page>
  );

  expect(container.firstChild).toMatchSnapshot();
});
