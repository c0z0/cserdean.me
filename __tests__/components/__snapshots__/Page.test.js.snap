// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot 1`] = `
<div
  class="sc-dnqmqq hpyIuR"
>
  <header>
    <div
      class="sc-bdVaJa cvLuQB"
    >
      <div
        class="sc-bwzfXH bsmKCX"
      >
        <a
          data-testid="menu-logo"
          href="/"
        >
          <img
            alt="logo"
            class="sc-htpNat cqNGZM"
            src="data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHdpZHRoPSIxNDE2IiBoZWlnaHQ9IjEyMjAiIHZpZXdCb3g9IjAgMCA3MDggNjEwIj4gIDxkZWZzPiAgICA8c3R5bGU+ICAgICAgLmNscy0xIHsgICAgICAgIGZpbGw6ICNDMUMxQzE7ICAgICAgfSAgICAgIC5jbHMtMSwgLmNscy0yLCAuY2xzLTMgeyAgICAgICAgZmlsbC1ydWxlOiBldmVub2RkOyAgICAgIH0gICAgICAuY2xzLTIgeyAgICAgICAgZmlsbDogI0QwRDBEMDsgICAgICB9ICAgICAgLmNscy0zIHsgICAgICAgIGZpbGw6ICNCNkI2QjY7ICAgICAgfSAgICA8L3N0eWxlPiAgPC9kZWZzPiAgPHBhdGggY2xhc3M9ImNscy0xIiBkPSJNNTg4Ljc4NiwyNTguOTY0bDUzLjQ0MSw5NS4yMDdMNDkxLjMzNCw2MTMuMzQySDk5NC4zMUw5NDQuMDEzLDcwNUgzNDAuNDQxWiIgdHJhbnNmb3JtPSJ0cmFuc2xhdGUoLTI4NyAtOTUpIi8+ICA8cGF0aCBjbGFzcz0iY2xzLTIiIGQ9Ik0yODcsNjEzLjM0MkwzNDAuNDQxLDcwNSw1ODguNzg2LDI2NS42NzNsMTUwLjg5MywyNjQuOGgxMDAuNkw1OTEuOTI5LDk1WiIgdHJhbnNmb3JtPSJ0cmFuc2xhdGUoLTI4NyAtOTUpIi8+ICA8cGF0aCBjbGFzcz0iY2xzLTMiIGQ9Ik01MzguNjQsNTMxLjE2Nkw0OTAuOCw2MTMuMzQySDk5NUw2OTguMTIyLDk1LjU0SDU5MS4yNEw4MzguNTA5LDUyOC4wMDVaIiB0cmFuc2Zvcm09InRyYW5zbGF0ZSgtMjg3IC05NSkiLz48L3N2Zz4="
            title="source"
          />
        </a>
        <button
          class="sc-bxivhb cGjZfZ"
          data-testid="menu-button"
          type="button"
        >
          <div
            class="sc-ifAKCX ewsDuX"
          >
            <div
              class="sc-EHOje gxQiPh"
            />
            <div
              class="sc-bZQynM hrtvGh"
            />
          </div>
        </button>
      </div>
      <div
        class="sc-gzVnrw jmEgEA"
        data-testid="menu-wrapper"
      >
        <div>
          <a
            class="sc-htoDjs cICBXG"
            data-testid="menu-item-target-/"
            href="/"
          >
            Home
          </a>
          <a
            class="sc-htoDjs bnKuMz"
            data-testid="menu-item-target-/projects"
            href="/projects"
          >
            Projects
          </a>
          <a
            class="sc-htoDjs bnKuMz"
            data-testid="menu-item-target-/ai"
            href="/ai"
          >
            AI Experiments
          </a>
        </div>
        <div>
          <a
            class="sc-htoDjs bnKuMz"
            data-testid="menu-item-target-/static/cv_cosmin_serdean.pdf"
            href="/static/cv_cosmin_serdean.pdf"
            rel="noopener noreferrer"
            target="_blank"
          >
            CV
          </a>
          <a
            class="sc-htoDjs bnKuMz"
            data-testid="menu-item-target-https://github.com/c0z0"
            href="https://github.com/c0z0"
            rel="noopener noreferrer"
            target="_blank"
          >
            GitHub
          </a>
          <a
            class="sc-htoDjs fAKAhJ"
            data-testid="menu-item-target-mailto:cosmoserdean@gmail.com"
            href="mailto:cosmoserdean@gmail.com"
            rel="noopener noreferrer"
            target="_blank"
          >
            Email
          </a>
        </div>
      </div>
    </div>
  </header>
  <div
    class="sc-iwsKbI ceQLln"
  >
    test content
  </div>
  <style>
    
        body {
          background: none;
        }
      
  </style>
</div>
`;

exports[`Matches snapshot 2`] = `
<div
  class="sc-dnqmqq cRwGLD"
>
  <header>
    <div
      class="sc-bdVaJa cvLuQB"
    >
      <div
        class="sc-bwzfXH bsmKCX"
      >
        <a
          data-testid="menu-logo"
          href="/"
        >
          <img
            alt="logo"
            class="sc-htpNat cqNGZM"
            src="data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHdpZHRoPSIxNDE2IiBoZWlnaHQ9IjEyMjAiIHZpZXdCb3g9IjAgMCA3MDggNjEwIj4gIDxkZWZzPiAgICA8c3R5bGU+ICAgICAgLmNscy0xIHsgICAgICAgIGZpbGw6ICNDMUMxQzE7ICAgICAgfSAgICAgIC5jbHMtMSwgLmNscy0yLCAuY2xzLTMgeyAgICAgICAgZmlsbC1ydWxlOiBldmVub2RkOyAgICAgIH0gICAgICAuY2xzLTIgeyAgICAgICAgZmlsbDogI0QwRDBEMDsgICAgICB9ICAgICAgLmNscy0zIHsgICAgICAgIGZpbGw6ICNCNkI2QjY7ICAgICAgfSAgICA8L3N0eWxlPiAgPC9kZWZzPiAgPHBhdGggY2xhc3M9ImNscy0xIiBkPSJNNTg4Ljc4NiwyNTguOTY0bDUzLjQ0MSw5NS4yMDdMNDkxLjMzNCw2MTMuMzQySDk5NC4zMUw5NDQuMDEzLDcwNUgzNDAuNDQxWiIgdHJhbnNmb3JtPSJ0cmFuc2xhdGUoLTI4NyAtOTUpIi8+ICA8cGF0aCBjbGFzcz0iY2xzLTIiIGQ9Ik0yODcsNjEzLjM0MkwzNDAuNDQxLDcwNSw1ODguNzg2LDI2NS42NzNsMTUwLjg5MywyNjQuOGgxMDAuNkw1OTEuOTI5LDk1WiIgdHJhbnNmb3JtPSJ0cmFuc2xhdGUoLTI4NyAtOTUpIi8+ICA8cGF0aCBjbGFzcz0iY2xzLTMiIGQ9Ik01MzguNjQsNTMxLjE2Nkw0OTAuOCw2MTMuMzQySDk5NUw2OTguMTIyLDk1LjU0SDU5MS4yNEw4MzguNTA5LDUyOC4wMDVaIiB0cmFuc2Zvcm09InRyYW5zbGF0ZSgtMjg3IC05NSkiLz48L3N2Zz4="
            title="source"
          />
        </a>
        <button
          class="sc-bxivhb cGjZfZ"
          data-testid="menu-button"
          type="button"
        >
          <div
            class="sc-ifAKCX bTSODv"
          >
            <div
              class="sc-EHOje gxQiPh"
            />
            <div
              class="sc-bZQynM hrtvGh"
            />
          </div>
        </button>
      </div>
      <div
        class="sc-gzVnrw jmEgEA"
        data-testid="menu-wrapper"
      >
        <div>
          <a
            class="sc-htoDjs hKRjuU"
            data-testid="menu-item-target-/"
            href="/"
          >
            Home
          </a>
          <a
            class="sc-htoDjs elorzL"
            data-testid="menu-item-target-/projects"
            href="/projects"
          >
            Projects
          </a>
          <a
            class="sc-htoDjs elorzL"
            data-testid="menu-item-target-/ai"
            href="/ai"
          >
            AI Experiments
          </a>
        </div>
        <div>
          <a
            class="sc-htoDjs elorzL"
            data-testid="menu-item-target-/static/cv_cosmin_serdean.pdf"
            href="/static/cv_cosmin_serdean.pdf"
            rel="noopener noreferrer"
            target="_blank"
          >
            CV
          </a>
          <a
            class="sc-htoDjs elorzL"
            data-testid="menu-item-target-https://github.com/c0z0"
            href="https://github.com/c0z0"
            rel="noopener noreferrer"
            target="_blank"
          >
            GitHub
          </a>
          <a
            class="sc-htoDjs zizbk"
            data-testid="menu-item-target-mailto:cosmoserdean@gmail.com"
            href="mailto:cosmoserdean@gmail.com"
            rel="noopener noreferrer"
            target="_blank"
          >
            Email
          </a>
        </div>
      </div>
    </div>
  </header>
  <div
    class="sc-iwsKbI ceQLln"
  >
    test content
  </div>
  <style>
    
        body {
          background: black;
        }
      
  </style>
</div>
`;
